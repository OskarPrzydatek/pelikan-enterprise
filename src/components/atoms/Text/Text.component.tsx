import * as S from './Text.styles';

interface IText {
  children: string | React.ReactNode;
  style?: React.CSSProperties | undefined;
  dataTestID?: string;
}

export const Text: React.FC<IText> = ({
  children,
  style,
  dataTestID,
}: IText) => {
  return (
    <S.Text data-testid={dataTestID} style={style}>
      {children}
    </S.Text>
  );
};
