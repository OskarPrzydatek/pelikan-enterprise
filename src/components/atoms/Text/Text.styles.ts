import styled from 'styled-components';

import { ICSS } from '~/models';

export const Text = styled.p<ICSS>`
  margin: 0;
  padding: 0;
  font-family: ${({ theme }) => theme.fonts.main};
  color: ${({ theme }) => theme.colors.black};
  font-size: ${({ theme }) => theme.fontSizes.s};

  ${({ css }) => css}

  @media ${({ theme }) => theme.breakpoints.phone} {
    font-size: ${({ theme }) => theme.fontSizes.s};
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    font-size: ${({ theme }) => theme.fontSizes.m};
  }

  @media ${({ theme }) => theme.breakpoints.desktop} {
    font-size: ${({ theme }) => theme.fontSizes.m};
  }
`;
