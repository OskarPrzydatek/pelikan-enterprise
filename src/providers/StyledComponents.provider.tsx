import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
