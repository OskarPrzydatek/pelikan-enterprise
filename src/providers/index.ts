export { StyledComponentsProvider } from './StyledComponents.provider';
