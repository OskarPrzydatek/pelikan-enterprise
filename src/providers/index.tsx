import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
