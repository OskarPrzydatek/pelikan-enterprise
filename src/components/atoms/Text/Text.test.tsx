import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { Text } from './Text';

const MockText = () => (
  <StyledComponentsProvider>
    <Text dataTestID="text-atom">Text</Text>
  </StyledComponentsProvider>
);

describe('Text', () => {
  test('component snapshot', () => {
    const view = render(<MockText />);
    expect(view).toMatchSnapshot();
  });

  test('ensure text children renders correctly', () => {
    render(<MockText />);
    const text = screen.getByTestId('text-atom');
    expect(text).toHaveTextContent('Text');
  });
});
