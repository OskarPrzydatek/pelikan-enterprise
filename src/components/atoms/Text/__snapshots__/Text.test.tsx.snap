// Vitest Snapshot v1

exports[`Text > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin: 0;
  padding: 0;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  font-size: 1rem;
}

@media (min-width:390px) {
  .c0 {
    font-size: 1.4rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 1.8rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 2rem;
  }
}

<body>
    <div>
      <p
        class="c0"
        data-testid="text-atom"
      >
        Text
      </p>
    </div>
  </body>,
  "container": .c0 {
  margin: 0;
  padding: 0;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  font-size: 1rem;
}

@media (min-width:390px) {
  .c0 {
    font-size: 1.4rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 1.8rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 2rem;
  }
}

<div>
    <p
      class="c0"
      data-testid="text-atom"
    >
      Text
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
